server/src/test/java/org/elasticsearch/index/query/IntervalQueryBuilderTests.java

```diff
@@ -114,7 +115,11 @@ private IntervalsSourceProvider createRandomSource(int depth) {
 
     private IntervalsSourceProvider.IntervalFilter createRandomFilter(int depth) {
         if (depth < 3 && randomInt(20) > 18) {
-            return new IntervalsSourceProvider.IntervalFilter(createRandomSource(depth + 1), randomFrom(filters));
+            if (randomBoolean()) {
+                return new IntervalsSourceProvider.IntervalFilter(createRandomSource(depth + 1), randomFrom(filters));
+            }
+            return new IntervalsSourceProvider.IntervalFilter(
+                new Script(ScriptType.INLINE, "mockscript", "1", Collections.emptyMap()));
```

Comment:
Adding these two variations is great to catch the xContent NPE but unfortunately doesn't directly trigger the Equals/Hashcode tests for the query builder due to a lack of coverage we have here (see https://github.com/elastic/elasticsearch/blob/master/test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java#L578). Without the equals/hashCode fixes this will trigger some of the "cachability" tests in this class, but very rarely.
I think adding these tests is out of scope for this PR but we should open an issue to maybe test wire- and xContent serialization of all the IntervalsSourceProvider implementations directly in unit tests. There seem to be none at the moment.

Reply:
Thanks @cbuescher , I opened https://github.com/elastic/elasticsearch/pull/49820 as a followup